### PR TITLE
DLL: More XInput stuff.

### PR DESCRIPTION
**Pull request by YaLTeR**
- Added XInput 1.4 hooks (1.3 ones + one extra).
- Added XInputGetStateEx() hook (unnamed, ordinal 100).
- Style fixes.
- Modified Xinput.h to contain 1.4 stuff and to only contain structs.
- Removed Xinput.lib for now (we don't link to it and it was most likely
  outdated anyway).
- Quick hack in intercept to allow hooking by unnamed ordinal.

### CODE REVIEW

File: external/Xinput.h [hunk 1]
```diff
@@ -163,121 +156,52 @@
...
 typedef struct _XINPUT_GAMEPAD
 {
-    WORD                                wButtons;
-    BYTE                                bLeftTrigger;
-    BYTE                                bRightTrigger;
-    SHORT                               sThumbLX;
-    SHORT                               sThumbLY;
-    SHORT                               sThumbRX;
-    SHORT                               sThumbRY;
+	WORD                                wButtons;
```
Comment (Warepire) [inline external/Xinput.h]: Why? The tabbing was correct before

File: external/Xinput.h [hunk 2]
```diff
@@ -163,121 +156,52 @@
...
+	SHORT                               sThumbLY;
+	SHORT                               sThumbRX;
+	SHORT                               sThumbRY;
 } XINPUT_GAMEPAD, *PXINPUT_GAMEPAD;
 
 typedef struct _XINPUT_STATE
 {
-    DWORD                               dwPacketNumber;
-    XINPUT_GAMEPAD                      Gamepad;
+	DWORD                               dwPacketNumber;
```
Comment (Warepire) [inline external/Xinput.h]: Here too

File: external/Xinput.h [hunk 3]
```diff
@@ -163,121 +156,52 @@
...
+	XINPUT_GAMEPAD                      Gamepad;
 } XINPUT_STATE, *PXINPUT_STATE;
 
 typedef struct _XINPUT_VIBRATION
 {
-    WORD                                wLeftMotorSpeed;
-    WORD                                wRightMotorSpeed;
+	WORD                                wLeftMotorSpeed;
```
Comment (Warepire) [inline external/Xinput.h]: Here too

File: external/Xinput.h [hunk 4]
```diff
@@ -163,121 +156,52 @@
...
 } XINPUT_VIBRATION, *PXINPUT_VIBRATION;
 
 typedef struct _XINPUT_CAPABILITIES
 {
-    BYTE                                Type;
-    BYTE                                SubType;
-    WORD                                Flags;
-    XINPUT_GAMEPAD                      Gamepad;
-    XINPUT_VIBRATION                    Vibration;
+	BYTE                                Type;
```
Comment (Warepire) [inline external/Xinput.h]: All of them actually ...

File: wintasee/hooks/Xinputhooks.cpp [hunk 5]
```diff
@@ -105,12 +120,12 @@ HOOKFUNC DWORD WINAPI MyXInputGetCapabilities(
      *
      * We set all bits to indicate maximal possible precision.
      */
-    supported_gamepad.bLeftTrigger = 0xFF;
-    supported_gamepad.bRightTrigger = 0xFF;
-    supported_gamepad.sThumbLX = -1;
-    supported_gamepad.sThumbLY = -1;
-    supported_gamepad.sThumbRX = -1;
-    supported_gamepad.sThumbRY = -1;
+    supported_gamepad.bLeftTrigger = static_cast<BYTE>(~0);
```
Comment (Warepire) [inline wintasee/hooks/Xinputhooks.cpp]: Unnecessary change, these casts can be implicit.

File: wintasee/hooks/Xinputhooks.cpp [hunk 6]
```diff
@@ -480,6 +495,47 @@ HOOKFUNC DWORD WINAPI MyXInputGetBatteryInformation(
...
+    DWORD  dwUserIndex,      // [in] Index of the gamer associated with the device.
+    LPWSTR pRenderDeviceId,  // [out, optional] Windows Core Audio device ID string for render (speakers).
+    UINT*  pRenderCount,     // [in, out, optional] Size, in wide-chars, of the render device ID string buffer.
+    LPWSTR pCaptureDeviceId, // [out, optional] Windows Core Audio device ID string for capture (microphone).
+    UINT*  pCaptureCount     // [in, out, optional] Size, in wide-chars, of capture device ID string buffer.
+)
+{
+    debuglog(LCF_JOYPAD, __FUNCTION__ "(dwUserIndex=%d) called.\n", dwUserIndex);
+
+    if (dwUserIndex < 0 || dwUserIndex > 3)
```
Comment (Warepire) [inline wintasee/hooks/Xinputhooks.cpp]: DWORDs are unsigned, cannot be less than 0.

File: wintasee/intercept.cpp [hunk 7]
```diff
@@ -148,6 +148,8 @@ struct InterceptAPIArgs
 	bool trampolineOnly;
 	const char* dllName;
 //	const char** dllNameArray; // if nonzero, we assume this is a dynamictramp hook and interpret the other arguments differently
+
```
Comment (Warepire) [inline wintasee/intercept.cpp]: Unnecessary blank line

File: wintasee/intercept.cpp [hunk 8]
```diff
@@ -148,6 +148,8 @@ struct InterceptAPIArgs
 	bool trampolineOnly;
 	const char* dllName;
 //	const char** dllNameArray; // if nonzero, we assume this is a dynamictramp hook and interpret the other arguments differently
+
+	LPCSTR ordinal;
```
Comment (Warepire) [inline wintasee/intercept.cpp]: Can the std::string be dropped in favor of a LPCSTR?

**Comment by YaLTeR** [comment]
Replying to all Xinput.h comments: my VS decided to switch back to tabs for no apparent reason and I didn't notice it until later. I'll fix it.

File: wintasee/intercept.cpp [hunk 9]
```diff
@@ -142,12 +142,13 @@ BOOL HookVTable(void* iface, int entry, FARPROC replace, FARPROC& oldfuncPointer
 #include <vector>
 struct InterceptAPIArgs
 {
-	std::string funcName;
+	LPCSTR funcName;
```
Comment (Warepire) [inline wintasee/intercept.cpp]: Please fix the spacing in the entire struct

File: wintasee/intercept.h [hunk 10]
```diff
@@ -228,6 +228,7 @@ struct InterceptDescriptor
 	FARPROC replacementFunction;
 	FARPROC trampolineFunction;
 	int enabled;
+	bool ordinal;
```
Comment (Warepire) [inline wintasee/intercept.h]: Same with this
